src/main.rs

```diff
@@ -4,33 +4,43 @@ extern crate easy_hash;
 
 use easy_hash::{Sha256, Hasher, HashResult};
 use std::thread;
-use std::sync::mpsc;
+use std::sync::{mpsc, Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
 
 const BASE: usize = 42;
 const THREADS: usize = 4;
 static DIFFICULTY: &'static str = "000000";
 
 struct Solution(usize, String);
 
-fn search_for_solution(start_at: usize, sender: mpsc::Sender<Option<Solution>>) {
+fn search_for_solution(start_at: usize, sender: mpsc::Sender<Solution>, is_solution_found: Arc<AtomicBool>) {
+    let mut iteration_no = 0;
```

Comment:
Here's an alternative proposal:

- Turn the for loop into a loop on iteration_no
- Explicitly compute the tried number as `number = iteration_no * THREADS * BASE`

This allows you to drop the step_by feature and build on stable.